packages/firestore/src/local/local_serializer.ts

```diff
@@ -56,7 +56,9 @@ export class LocalSerializer {
     } else if (remoteDoc.noDocument) {
       const key = DocumentKey.fromSegments(remoteDoc.noDocument.path);
       const version = this.fromDbTimestamp(remoteDoc.noDocument.readTime);
-      return new NoDocument(key, version);
+      return new NoDocument(key, version, {
```

Comment:
DbRemoteDocument is the persisted version of the document. I felt that it was saner to directly encode the boolean flag there, especially since the schema is already in JSON and has the name encoded.

DbNoDocument looks this way to match DbDocument.